lang/src/org/partiql/lang/syntax/PartiQLParser.kt

```diff
@@ -184,4 +190,33 @@ internal class PartiQLParser(
             throw ParserException(message = msg, errorCode = ErrorCode.PARSE_UNEXPECTED_TOKEN, errorContext = propertyValues, cause = e)
         }
     }
+
+    /**
+     * A wrapped [GeneratedParser] to allow thread interruption during parse.
+     */
+    internal class InterruptibleParser(input: TokenStream) : GeneratedParser(input) {
+        override fun enterRule(localctx: ParserRuleContext?, state: Int, ruleIndex: Int) {
+            checkThreadInterrupted()
+            super.enterRule(localctx, state, ruleIndex)
+        }
+    }
+
+    /**
+     * This token stream creates [parameterIndexes], which is a map with key-value pairs, where the keys represent the
```

Comment:
nit: redundant description

```suggestion
     * This token stream creates [parameterIndexes], which is a map, where the keys represent the
```